### PR TITLE
fix(ethapi): make `eth_getTransactionReceipt` null for unexecuted/unknown transactions

### PR DESCRIPTION
**Pull request by cryptoAtwill**
This one addresses #1003. 

From the issue above, it seems the txn hash is found but the block information is not found. This could be the latest block is not yet confirmed. This PR checks if block header is found when txn hash is present, it will return None. Else if the block header is found but block data is not found, it will throw an error.

### CODE REVIEW

File: fendermint/eth/api/src/apis/eth.rs [hunk 1]
```diff
@@ -484,32 +484,39 @@ pub async fn get_transaction_receipt<C>(
...
+            &header.header,
+            &state_params.value.base_fee,
+        )
+        .await
+        .context("failed to convert to receipt")?;
+
+        Ok(Some(receipt))
     } else {
-        Ok(None)
+        error(ExitCode::USR_ILLEGAL_ARGUMENT, "incompatible transaction")
```
[the remark this answers is not in the document]
Comment (raulk) [reply fendermint/eth/api/src/apis/eth.rs:519]: Do those not have a receipt we can return?

File: fendermint/eth/api/src/apis/eth.rs [hunk 2]
```diff
@@ -484,32 +484,39 @@ pub async fn get_transaction_receipt<C>(
...
-            let receipt = to_eth_receipt(
-                &msg,
-                &res,
-                &cumulative,
-                &header.header,
-                &state_params.value.base_fee,
-            )
-            .await
-            .context("failed to convert to receipt")?;
+    let Some(res) = data.tx_by_hash(tx_hash).await? else {
```
Comment (raulk) [inline fendermint/eth/api/src/apis/eth.rs:487]: Can we call this var tx?

**Review by raulk — COMMENTED** [review]
Is there a test you can augment with these scenarios?

File: fendermint/eth/api/src/apis/eth.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (cryptoAtwill) [reply fendermint/eth/api/src/apis/eth.rs:519]: The other kind of message is IPC mesage, which I believe the receipts are not handled like signed txns, we might not get a proper txn receipt. Or we can just return `None` regardless of whatever error and just log these error using metrics

**Review by raulk — APPROVED** [review]
@cryptoAtwill We don't seem to have structured integration tests for the Eth API, so covering this in a test case is a non-starter for now. Did you successfully test this manually?

**Comment by raulk** [comment]
Pending test results from @cryptoAtwill.

**Comment by bitSmiley** [comment]
I used a simple script to test this:
```bash
RPC_URL="http://localhost:8545"

txn_hash=$(cast send --rpc-url ${RPC_URL} --private-key <SECRET> --value 1gwei <ADDRESS> --async)

echo "${txn_hash}"

curl ${RPC_URL} \
  -X POST \
  -H "Content-Type: application/json" \
  --data "{\"method\":\"eth_getTransactionReceipt\",\"params\":[\"${txn_hash}\"],\"id\":1,\"jsonrpc\":\"2.0\"}"
```

The result I got is:
```
{"jsonrpc":"2.0","result":null,"id":1}
```

It should be fine.

**Comment by cryptoAtwill** [comment]
I have tested this using:
```
RPC_URL="http://localhost:8545"
txn_hash=$(cast send --rpc-url ${RPC_URL} --private-key <PRIVATE_KEY> --value 1gwei <ADDRESS> --async)
echo "${txn_hash}"

curl ${RPC_URL} \
  -X POST \
  -H "Content-Type: application/json" \
  --data "{\"method\":\"eth_getTransactionReceipt\",\"params\":[\"${txn_hash}\"],\"id\":1,\"jsonrpc\":\"2.0\"}"
```

The result I got: {"jsonrpc":"2.0","result":null,"id":1}. I think it will only be triggered under certain race condition, this PR should not have caused extra errors.

**Pull request merged**